### PR TITLE
Training with multiscale discrims and featmatch

### PR DESCRIPTION
**Pull request by 51N84D**
Adding mutli-scale discriminators, and feature-matching loss

### CODE REVIEW

File: omnigan/blocks.py [hunk 1]
```diff
@@ -372,27 +371,24 @@ def __init__(
...
         Args:
-            latent_shape (tuple): z's shape (only the number of channels matters)
+            latent_dim (tuple): z's shape (only the number of channels matters)
             cond_nc (int): conditioning tensor's expected number of channels
             spade_n_up (int): Number of total upsamplings from z
             spade_use_spectral_norm (bool): use spectral normalization?
             spade_param_free_norm (str): norm to use before SPADE de-normalization
             spade_kernel_size (int): SPADE conv layers' kernel size
-
+            fullspade (bool): Produces an image as a final  result
```
Comment (alexrey88) [inline omnigan/blocks.py:391]: Where is this parameter used?

File: omnigan/norms.py [hunk 2]
```diff
@@ -83,14 +83,13 @@ def l2normalize(v, eps=1e-12):
...
-    and the Pytorch implementation
-    https://github.com/christiancosgrove/pytorch-spectral-normalization-gan
+    Based on the paper "Spectral Normalization for Generative Adversarial Networks" by Takeru Miyato, 
+    Toshiki Kataoka, Masanori Koyama, Yuichi Yoshida and the 
+    Pytorch implementation https://github.com/christiancosgrove/pytorch-spectral-normalization-gan
     """
 
     def __init__(self, module, name="weight", power_iterations=1):
-        super().__init__()
+        super(SpectralNorm, self).__init__()
```
Comment (vict0rsch) [inline omnigan/norms.py:92]: in python 3 it's ok to just do `super().__init__()`

File: shared/trainer/defaults.yaml [hunk 3]
```diff
@@ -118,6 +119,8 @@ dis:
     init_type: kaiming
     init_gain: 0.2
     use_sigmoid: false
+    num_D: 1 #Number of discriminators to use (>1 means multi-scale)
```
Comment (vict0rsch) [inline shared/trainer/defaults.yaml:122]: don't we want `3` as default? is 1 enough?

File: omnigan/blocks.py (hunk 1 above)
Comment (51N84D) [reply omnigan/blocks.py:391]: It's not. Good catch, I'll get rid of it

File: shared/trainer/defaults.yaml (hunk 3 above)
Comment (51N84D) [reply shared/trainer/defaults.yaml:122]: Sure, I just didn't want to make the choice on everyone's behalf ;) I'll change it

**Pull request merged**